java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/ShareFileService.java

```diff
@@ -31,64 +30,77 @@
 
 
 @Service
-public class ShareGpxService {
+public class ShareFileService {
 
 	@Autowired
 	protected PremiumUserFilesRepository filesRepository;
 
+	@Autowired
+	protected ShareFileRepository shareFileRepository;
+
 	@Autowired
 	protected PremiumUsersRepository usersRepository;
 
 	@Autowired
 	UserdataService userdataService;
 
-	protected static final Log LOGGER = LogFactory.getLog(ShareGpxService.class);
-
-	private static final String BLACK_LIST = "blacklist";
-	private static final String WHITE_LIST = "whitelist";
+	protected static final Log LOGGER = LogFactory.getLog(ShareFileService.class);
 
 	private static final String SEPARATOR = ",";
+	private static final String CODE_SEPARATOR = "-";
+	public static final int CODE_LENGTH = 8;
 
 	Gson gson = new Gson();
 
 	@Transactional
 	public String generateSharedCode(PremiumUserFilesRepository.UserFile userFile, String type, @Nullable String groupActions) {
-		String uniqueToken = UUID.randomUUID().toString();
-		userFile.sharedCode = uniqueToken;
-		userFile.sharedInfo = addSharedInfo(type, groupActions);
-		filesRepository.saveAndFlush(userFile);
+		String uniqueCode = UUID.randomUUID().toString().substring(0, CODE_LENGTH);
```

Comment:
UUID using CODE_LENGTH=8 limits range to 0xFFFFFFFF (4 billion shares max). Could be not enough for unlimited cloud storage and not enough against brute force. But OK for now.